### PR TITLE
feat: add option to set debug log level

### PR DESCRIPTION
**Pull request by JoshuaMoelans**
Fixes https://github.com/getsentry/sentry-native/issues/888
Related to https://github.com/getsentry/sentry-cocoa/issues/4467

The main goal is to allow for outputting just the warnings from `sentry_init`. To accommodate this, we introduce log levels as a way to control what level of output the user gets. We decided against splitting up these levels into `init`-only and global levels, since this doesn't seem like a necessary feature as of yet.

### CODE REVIEW

File: src/sentry_options.c [hunk 1]
```diff
@@ -396,6 +401,20 @@ sentry_options_set_logger(
     opts->logger.logger_data = userdata;
 }
 
+void
+sentry_options_set_logger_level(sentry_options_t *opts, sentry_level_t level)
+{
+    opts->logger.logger_level = level;
+    opts->debug = true;
```
Comment (JoshuaMoelans) [inline src/sentry_options.c:408]: We implicitly expect that the user wants to have debug output when setting the logger level (hence we call this from `sentry_options_set_debug` as well).

**Review by supervacuus — CHANGES_REQUESTED** [review]
I expected the macro name change to highlight a few inconsistencies.

A short sync might also be acceptable regarding the options interface.

File: include/sentry.h [hunk 2]
```diff
@@ -1078,6 +1078,18 @@ SENTRY_API void sentry_options_set_debug(sentry_options_t *opts, int debug);
...
+ */
+SENTRY_API void sentry_options_set_logger_level(
+    sentry_options_t *opts, sentry_level_t level);
+
+/**
+ * Sets the log level for the sentry_init function
+ */
+SENTRY_API void sentry_options_set_init_log_level(
+    sentry_options_t *opts, sentry_level_t level);
+
```
[the remark this answers is not in the document]
Comment (JoshuaMoelans) [reply include/sentry.h]: Removed this overcomplication splitting up global and init log levels. Now we just allow setting of the log level globally.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,10 @@
 # Changelog
 
+## Unreleased
+
+**Features**:
+- Add `sentry_init` and global logger log levels ([#1107](https://github.com/getsentry/sentry-native/pull/1107))
```
Comment (supervacuus) [inline CHANGELOG.md:6]:
```suggestion
**Features**:

- Add option to set debug log level. ([#1107](https://github.com/getsentry/sentry-native/pull/1107))
```

**Review by supervacuus — APPROVED** [review]

**Pull request merged**